job/fetch.go

```diff
@@ -12,16 +12,16 @@ var (
 	servers singleflight.Group
 )
 
-func discoverHardwareFromDHCP(ctx context.Context, mac net.HardwareAddr, giaddr net.IP, circuitID string) (packet.Discovery, error) {
+func discoverHardwareFromDHCP(ctx context.Context, mac net.HardwareAddr, giaddr net.IP, circuitID string) (context.Context, packet.Discovery, error) {
 	fetch := func() (interface{}, error) {
 		return client.DiscoverHardwareFromDHCP(ctx, mac, giaddr, circuitID)
 	}
 	v, err := servers.Do(mac.String(), fetch)
 	if err != nil {
-		return nil, err
+		return ctx, nil, err
```

Comment:
can we return nil for the context instead, it just feels simpler. The caller won't have to worry about doing anything with the context being returned and can just keep it out of mind.

Reply:
> can we return nil for the context instead, it just feels simpler. The caller won't have to worry about doing anything with the context being returned and can just keep it out of mind.

I do not think we should do that. While boots might usually be done with a context when there is an error, that's not a reason to break the context chain. Especially so when we're doing otel things through context. Nobody ever checks ctx for nil, and they shouldn't have to.

Reply:
Actually I don't see why we need to return the context here at all. Its never being mutated and thus the return is only confusing right?

Reply:
Yep, I traced back through the code paths and don't remember why I even added it. Probably started thinking I'd need it then found the other path in job.setup. Commit coming soon to remove.